### PR TITLE
S3t ingredient gateway

### PR DESCRIPTION
**Pull request by jasonrees1**
Added ingredient gateway class and interface.
Ingredient gateway formats ingredients from array into URL, then sends get request to API and receives recipes.

### CODE REVIEW

File: src/Ingredients/IngredientGateway.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+use BreadAndIfit\Interfaces\GatewayInterface;
+class IngredientGateway implements GatewayInterface
+{
+    static private function formatData(array $ingredients): string
+    {
+        $url = "http://www.recipepuppy.com/api/?i=";
+        foreach ($ingredients as $ingredient) {
+            $url .= $ingredient;
+            $url .= ',';
```
Comment (harrycoulton) [inline src/Ingredients/IngredientGateway.php]: These can be condensed down into one - you can just concatenate the comma onto the end: $ingredient.',';

File: src/Ingredients/IngredientGateway.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        }
+        curl_close($ch);
+        return $return;
+    }
+
+    static public function sendDataReturnResponse(array $ingredients): string
+    {
+        $recipes = self::sendDataToAPI(self::formatData($ingredients));
+        return $recipes;
+
```
Comment (harrycoulton) [inline src/Ingredients/IngredientGateway.php]: whitespace?

**Review by harrycoulton — COMMENTED** [review]
Can't see many major issues, the only issues I saw are the two minors highlighted below. 

Would recommend reducing all the unnecessary whitespace throughout.

File: src/Ingredients/IngredientGateway.php (hunk 1 above)
Comment (avocadobelly) [reply src/Ingredients/IngredientGateway.php]: done :)

**Review by harrycoulton — APPROVED** [review]
All looks PEACHY to me

**Pull request merged**